### DISCUSSION TITLE
Scheduling oddity

### DISCUSSION THREAD

**Discussion by aptanet** [containrrr/watchtower#1558] [Q&A]
I'm not sure whether I'm misunderstanding the scheduling parameters or what is going on here, but I'm finding that Watchtower seems to run multiple times on each scheduled run. If I'm using just a daily run, for example, it is fine (so not a show stopping bug if it is one), but in testing I've tried running every five or ten minutes and that syntax causes issues.

So, when used in an an environment section in a Docker compose file...

This works fine:
      WATCHTOWER_SCHEDULE: "0 45 14 * * *" # Everyday at 14:45

This runs 10 times every 10 minutes:
      WATCHTOWER_SCHEDULE: "* */10 * * * *" # Every 5 mins

This runs 20 times every 5 minutes starting at 2 minutes past the hour:
      WATCHTOWER_SCHEDULE: "* 2/5 * * * *" # Every 5 mins

This runs 10 times every 5 minutes starting at 1 minute past the hour:
      WATCHTOWER_SCHEDULE: "* 1/5 * * * *" # Every 5 mins

It isn't a setting that I'm likely to keep using, but it is puzzling me enough to mention it. This is happening with both the Discord and email notifications.

My full Docker compose file is (obfuscated obviously):
version: "3.9"
services:
  watchtower:
    image: containrrr/watchtower
    container_name: watchtower
    hostname: myserver
    environment:
      WATCHTOWER_MONITOR_ONLY: 'true'
      WATCHTOWER_SCHEDULE: "* 1/5 * * * *" # Every 5 mins
      WATCHTOWER_NOTIFICATIONS: 'email shoutrrr'
      WATCHTOWER_NOTIFICATION_EMAIL_FROM: paul@example.com
      WATCHTOWER_NOTIFICATION_EMAIL_TO: paul@example.net
      WATCHTOWER_NOTIFICATION_EMAIL_SERVER: smtp.example.com
      WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PORT: 25
      WATCHTOWER_NOTIFICATION_URL: discord://longrandomstrigofcharacters@0123456890
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    restart: unless-stopped

**Comment by piksel** [comment]
That does seem odd. Could you post logs of it running? It should show how it interprets the schedule. The docs for `/` is a bit unclear to me not, but the 2/5 should obviously not run 20 times at once.

**Reply by aptanet** [reply]
Not to worry, it is my 25 years (feeling old now) of working with cron completely failing to kick in (even allowing for the slightly different format to what I usually use). The first value is a *, therefore will match every second. If I'd had 60 responses then I may have twigged, but I suspect the frequency just overloaded something and it just did what it could manage somehow. The first value should be numeric to avoid that. A 0 makes sense, but any value would work to run on whichever second of the minute you want. The 2/5 starts on the second minute past the hour and then runs every 5 minutes after that. 2-37/5 would run every 5 minutes from 2 minutes past the hour to 37 minutes past (inclusive) - when used in the minutes column of course. The example that works has the 0 in the first value for the seconds - typical! I'm going to put it down to other things going on in my life at the moment (being a carer as well as working) and hope this question doesn't come up in Google too often when I'm searched for ;-)
